packages/web-pkg/src/composables/resourcesViewDefaults/useResourcesViewDefaults.ts

```diff
@@ -0,0 +1,119 @@
+import { nextTick, computed, unref, Ref } from 'vue'
+import { folderService } from '../../services/folder'
+import { fileList } from '../../helpers/ui'
+import { usePagination, useSort, SortDir, SortField, useRouteName } from 'web-pkg/src/composables'
+import { useSideBar } from 'web-pkg/src/composables/sideBar'
+
+import {
+  queryItemAsString,
+  useMutationSubscription,
+  useRouteQuery,
+  useStore
+} from 'web-pkg/src/composables'
+import { determineSortFields as determineResourceTableSortFields } from '../../helpers/ui/resourceTable'
+import { determineSortFields as determineResourceTilesSortFields } from '../../helpers/ui/resourceTiles'
+import { Task } from 'vue-concurrency'
+import { Resource } from 'web-client'
+import { useSelectedResources, SelectedResourcesResult } from '../selection'
+import { ReadOnlyRef } from 'web-pkg'
+import {
+  useFileListHeaderPosition,
+  useViewMode,
+  useViewSize,
+  ViewModeConstants
+} from 'web-pkg/src/composables'
+
+import { ScrollToResult, useScrollTo } from '../scrollTo'
+
+interface ResourcesViewDefaultsOptions<T, U extends any[]> {
+  loadResourcesTask?: Task<T, U>
+}
+
+type ResourcesViewDefaultsResult<T, TT, TU extends any[]> = {
+  fileListHeaderY: Ref<any>
+  refreshFileListHeaderPosition(): void
+  loadResourcesTask: Task<TT, TU>
+  areResourcesLoading: ReadOnlyRef<boolean>
+  storeItems: ReadOnlyRef<T[]>
+  sortFields: ReadOnlyRef<SortField[]>
+  paginatedResources: Ref<T[]>
+  paginationPages: ReadOnlyRef<number>
+  paginationPage: ReadOnlyRef<number>
+  handleSort({ sortBy, sortDir }: { sortBy: string; sortDir: SortDir }): void
+  sortBy: ReadOnlyRef<string>
+  sortDir: ReadOnlyRef<SortDir>
+  viewMode: ReadOnlyRef<string>
+  viewSize: ReadOnlyRef<number>
+  selectedResources: Ref<Resource[]>
+  selectedResourcesIds: Ref<(string | number)[]>
+  isResourceInSelection(resource: Resource): boolean
+
+  sideBarOpen: Ref<boolean>
+  sideBarActivePanel: Ref<string>
+} & SelectedResourcesResult &
+  ScrollToResult
+
+export const useResourcesViewDefaults = <T, TT, TU extends any[]>(
+  options: ResourcesViewDefaultsOptions<TT, TU> = {}
+): ResourcesViewDefaultsResult<T, TT, TU> => {
+  const loadResourcesTask = options.loadResourcesTask || folderService.getTask()
```

Comment:
I don't think it's a really good idea to move the whole FolderLoaderService and all of the loaders into web-pkg at this point.

I would suggest to make `options.loadResourcesTask` of type: ` Task<TT, TU> || () =>  Task<TT, TU>`
and then pass it in everywhere where useResourcesViewDefaults is used: `{ loadResourcesTask: () => folderService.getTask() } `

Reply:
Looked at it, the main issue is that the folderService is located in web-app-files. Which shouldn't be reachable from web-app-backups

Reply:
Yes, that's why I said you could inject a getTask function...

Reply:
My main issue is that there are so many route checks in the loaders, so they are actually pretty files app specific...

Reply:
I might have explained poorly.
`packages/web-app-backups/src/components/BackupsSection.vue
` line: 141

uses `useResourcesViewDefaults`, which then should get the injected `{ loadResourcesTask: () => folderService.getTask() }`, the issue here is that for `folderService` to be injected, it would have to have access to _web-app-files_, which goes against the goal of this issue, no?

Reply:
Aaah, the app itself needs the loaders? Grmpfmhl

Reply:
Okay so turns out the backups app only uses the `useResourcesViewDefaults` composable to get `selectedResourcesIds`. Hence we suggest to leave the `useResourcesViewDefaults` and all the loaders in the files-app and only move the `useSelectedResources` composable to web-pkg.

Reply:
Done, but I could have missed unnecessary files that could be removed as well.